test/integration/framework/master_utils.go

```diff
@@ -126,40 +127,52 @@ func startMasterOrDie(masterConfig *master.Config) (*master.Master, *httptest.Se
 	}))
 
 	if masterConfig == nil {
-		etcdClient := NewEtcdClient()
-		storageVersions := make(map[string]string)
-		etcdStorage, err := master.NewEtcdStorage(etcdClient, latest.GroupOrDie("").InterfacesFor, latest.GroupOrDie("").GroupVersion.String(), etcdtest.PathPrefix())
-		storageVersions[""] = latest.GroupOrDie("").GroupVersion.String()
-		if err != nil {
-			glog.Fatalf("Failed to create etcd storage for master %v", err)
-		}
-		expEtcdStorage, err := NewExtensionsEtcdStorage(etcdClient)
-		storageVersions["extensions"] = testapi.Extensions.GroupAndVersion()
-		if err != nil {
-			glog.Fatalf("Failed to create etcd storage for master %v", err)
-		}
-		storageDestinations := master.NewStorageDestinations()
-		storageDestinations.AddAPIGroup("", etcdStorage)
-		storageDestinations.AddAPIGroup("extensions", expEtcdStorage)
-
-		masterConfig = &master.Config{
-			StorageDestinations:  storageDestinations,
-			StorageVersions:      storageVersions,
-			KubeletClient:        kubeletclient.FakeKubeletClient{},
-			EnableLogsSupport:    false,
-			EnableProfiling:      true,
-			EnableSwaggerSupport: true,
-			EnableUISupport:      false,
-			APIPrefix:            "/api",
-			APIGroupPrefix:       "/apis",
-			Authorizer:           apiserver.NewAlwaysAllowAuthorizer(),
-			AdmissionControl:     admit.NewAlwaysAdmit(),
-		}
+		masterConfig = NewMasterConfig()
+		masterConfig.EnableProfiling = true
+		masterConfig.EnableSwaggerSupport = true
 	}
 	m = master.New(masterConfig)
 	return m, s
 }
 
+// Returns a basic master config.
+func NewMasterConfig() *master.Config {
+	etcdClient := NewEtcdClient()
+	storageVersions := make(map[string]string)
+	etcdStorage, err := master.NewEtcdStorage(etcdClient, latest.GroupOrDie("").InterfacesFor, latest.GroupOrDie("").GroupVersion.String(), etcdtest.PathPrefix())
```

Comment:
Use testapi for the GroupVersion?

IIRC, using latest to get InterefacesFor is inevitable, since the testapi doesn't export InterfacesFor.